File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque
def solution(people, limit):
    answer = 0
    
    people.sort(reverse= True)
    
    people = deque(people)
    i = 0
    while len(people)>1:
        if len(people) > 1 :
            if people[0] + people[-1] <= limit:
                people.popleft()
                people.pop()
                answer+=1

            else:
                people.popleft()
                answer+=1
        
    if len(people) == 1:
        answer+=1
            
        
    
    
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Integer[] sp = new Integer[people.length];
        
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        
        for(int i=0; i<people.length;i++){
            sp[i] = people[i];
        }
        Arrays.sort(sp,Collections.reverseOrder());
        
        for(int p:sp){
            if(pq.isEmpty()){
                if((limit-p)>=40)pq.add(limit-p);
                answer++;
            }else{
                if(pq.peek()>=p){
                    pq.poll();
                }else{
                    if((limit-p)>=40)pq.add(limit-p);
                    answer++;
                }
            }
        }
        
        
        return answer;
    }
}